### ISSUE TITLE
Sh-elf-gcc Missing

### ISSUE THREAD

**Issue by codepoet80** [TimSimpson/KosVM#2]
Ran your script successfully (thanks so much for making it) in Cygwin on both Win10 and Win7. In both cases, upon completion, I ran the command "source environ.sh" and got an error that sh-elf-gcc is missing...

```
$ source environ.sh
/home/Jon/dc/kos/utils/gnu_wrappers/kos-cc: line 34: /opt/toolchains/dc/sh-elf/bin/sh-elf-gcc: No such file or directory
```

Tried it on two completely different computers, with different OSes, with Cygwin installed as instructed for Windows. The problem is fatal and reproduced every time.

**Comment by TimSimpson** [comment]
Interesting. I have not worked with the Dreamcast for several years but
recently have built a tool chain using this script and Bash for Ubuntu for
Windows. So far I've been able to build executables from the examples
though I haven't been able to run it yet.

I'll attempt building this on Cygwin soon and see what the issue is, though
I've been so impressed with Ubuntu on Windows I may switch to using it
primarily.

Thanks for reporting this problem. If it persists I'll update the readme so
I won't lead anyone else astray.

-Tim

On Oct 1, 2016 2:35 PM, "codepoet80" notifications@github.com wrote:

Ran your script successfully (thanks so much for making it) in Cygwin on
both Win10 and Win7. In both cases, upon completion, I ran the command
"source environ.sh" and got an error that sh-elf-gcc is missing...

$ source environ.sh
/home/Jon/dc/kos/utils/gnu_wrappers/kos-cc: line 34:
/opt/toolchains/dc/sh-elf/bin/sh-elf-gcc: No such file or directory

Tried it on two completely different computers, with different OSes, with
Cygwin installed as instructed for Windows. The problem is fatal and
reproduced every time.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
https://github.com/TimSimpson/KosVM/issues/2, or mute the thread

**Comment by codepoet80** [comment]
i'll try Bash for Windows. Also downloading an Ubuntu image to give that a try...

**Comment by codepoet80** [comment]
In Bash for Windows 10, the install script completed, but I am unable to "make" any of the examples.
Stepping through the install script it looks like the build_kos function is failing to "make" with the error:
`/opt/toolchains/dc/arm-eabi/bin/arm-eabi-gcc: Command not found`

Full log attached. I confirmed that the file arm-eabi-gcc is not in the expected directory.

[build_kos-make_errors.txt](https://github.com/TimSimpson/KosVM/files/504955/build_kos-make_errors.txt)

I put a "pause" (wait for key stroke) in after every major functional call, I'll try to find out where the root cause is occurring...

**Comment by TimSimpson** [comment]
Try this- assuming you already ran install_kos.sh try compiling the PVR
examples located somewhere in ~/Tools/Dreamcast/examples (the specific one
is called pvr_mark_strips_direct).

If that works then the problem may be that install_kos.sh is not running
all of the targets. I wrote it years ago when I was having a hell of a time
getting the tool chain to build and made it call make several times for
different targets instead of just building everything, so I could more
easily keep track of what I was doing. Possibly more targets got added
since then, such as the compiler for the sound processor. Or maybe I never
built that correctly in the first place. :)

Either way if this is what is happening I'll be glad to fix it.

**Comment by codepoet80** [comment]
yes, that's what i tried. i'm unable to compile any of the samples (that particular sample is _pvrmark_strips_direct_ -- you have an extra underscore both here and in the documentation) because of the missing arm-eabi-gcc file.

the final output when running your install script, which is fatal, and suggests that the script did not finish is:

```
make[6]: Leaving directory `/home/Jon/Tools/dreamcast/kallistios/utils/dc-chain/build-gcc-arm-eabi-4.7.3/arm-eabi/thumb/libgcc'
make[5]: Leaving directory `/home/Jon/Tools/dreamcast/kallistios/utils/dc-chain/build-gcc-arm-eabi-4.7.3/arm-eabi/thumb/libgcc'
make[5]: write error
make[4]: *** [multi-do] Error 1
make[4]: Leaving directory `/home/Jon/Tools/dreamcast/kallistios/utils/dc-chain/build-gcc-arm-eabi-4.7.3/arm-eabi/libgcc'
make[3]: *** [all-multi] Error 2
make[3]: Leaving directory `/home/Jon/Tools/dreamcast/kallistios/utils/dc-chain/build-gcc-arm-eabi-4.7.3/arm-eabi/libgcc'
make[3]: write error
make[2]: *** [all-target-libgcc] Error 1
make[2]: Leaving directory `/home/Jon/Tools/dreamcast/kallistios/utils/dc-chain/build-gcc-arm-eabi-4.7.3'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/home/Jon/Tools/dreamcast/kallistios/utils/dc-chain/build-gcc-arm-eabi-4.7.3'
make: *** [build-arm-gcc-pass1] Error 1
```

I believe the crash is happening during the **build_arm_tools** function of your script, since I saw the previous function **build_sh4_tools** complete successfully. The environ.sh script is **not** created, so the crash happens somewhere between line 240 and line 242 of your script. Hope that helps troubleshoot.

You're right, building this toolchain is a royal pain. I really appreciate your script, and willingness to update for whatever has changed!

**Comment by codepoet80** [comment]
I tried your script on an actual Ubuntu 64-bit image. The install got significantly further, but there was a failure at the end. Here's the tail of the script output. I'll try to figure out how to fix manually...

```
zlib
+ [[ zlib == \o\p\u\s ]]
+ [[ zlib == \o\p\u\s\f\i\l\e ]]
+ [[ zlib == \s\c\r\i\p\t\s ]]
+ [[ zlib == \u\t\i\l\s ]]
+ echo 'Buidling zlib.'
Buidling zlib.
+ pushd zlib
~/Tools/dreamcast/kos-ports/zlib ~/Tools/dreamcast/kos-ports ~/KosVM
+ make
Please build your port with 'make install clean'.
+ popd
~/Tools/dreamcast/kos-ports ~/KosVM
+ popd
~/KosVM
+ build_dc_tool
+ pushd /home/jon/Tools/dreamcast
~/Tools/dreamcast ~/KosVM
+ git clone git://cadcdev.git.sourceforge.net/gitroot/cadcdev/dcload-ip
Cloning into 'dcload-ip'...
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

```
